### PR TITLE
api doctor dotnet core migration and replaces windows by ubuntu

### PR DESCRIPTION
**Pull request by baywet**

### CODE REVIEW

File: azure-pipelines/e2e-templates/snippet-generation.yml [hunk 1]
```diff
@@ -2,41 +2,35 @@
...
+    # release folder can change based on .NET core version, so search recursively in bin folder
+    $snippetGeneratorBinDirectory = Join-Path $env:BUILD_SOURCESDIRECTORY/microsoft-graph-explorer-api/CodeSnippetsReflection.App/bin $(buildConfiguration)
+    $snippetGeneratorPath = (Get-ChildItem $snippetGeneratorBinDirectory *App.exe -Recurse).FullName
+    Write-Host "Path to snippet generator tool: $snippetGeneratorPath"
 
-      $apidoctorBinDirectory = Join-Path $env:Build_SourcesDirectory\apidoctor\ApiDoctor.Console\bin $(buildConfiguration)
-      $apidoctorPath = (Get-ChildItem $apidoctorBinDirectory apidoc.exe -Recurse).FullName
-      Write-Host "Path to apidoctor tool: $apidoctorPath"
+    $apidoctorBinDirectory = Join-Path $env:BUILD_SOURCESDIRECTORY/apidoctor/ApiDoctor.Console/bin $(buildConfiguration)
+    $apidoctorPath = (Get-ChildItem $apidoctorBinDirectory apidoc.exe -Recurse).FullName
```
Comment (zengin) [inline azure-pipelines/e2e-templates/snippet-generation.yml:31]: .exe

Comment (baywet) [reply azure-pipelines/e2e-templates/snippet-generation.yml:31]: I'm not sure I understand the comment here?

File: msgraph-sdk-raptor-compiler-lib/appsettings.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
   "ConnectionStrings": {
     "Raptor": "Server=Server;Database=Raptor;User Id=user;Password=pwd;"
   },
-  "IsLocalRun": true,
+  "IsLocalRun": false,
```
Comment (zengin) [inline msgraph-sdk-raptor-compiler-lib/appsettings.json]: I accidently checked in this file a couple of times before to a remote branch. Opened this issue to track a better management strategy for settings:
https://github.com/microsoftgraph/msgraph-sdk-raptor/issues/135

File: azure-pipelines/e2e-templates/snippet-generation.yml (hunk 1 above)
Comment (zengin) [reply azure-pipelines/e2e-templates/snippet-generation.yml:31]: Does it generate an apidoc.exe in linux?

Comment (baywet) [reply azure-pipelines/e2e-templates/snippet-generation.yml:31]: yep, as linux doesn't care so much about file extensions. This way you have a consistent experience with the output of a build between linux and windows. I'm guessing they did that to better support containers that are cross OS.

Comment (zengin) [reply azure-pipelines/e2e-templates/snippet-generation.yml:26]: That is interesting, you have changed it to apidoc.exe to apidoc here: https://github.com/microsoftgraph/microsoft-graph-explorer-api/pull/387/files#diff-828747545144203996dea461f81a0453af29635c4c8b731f0523109ad679e76eR103

Comment (baywet) [reply azure-pipelines/e2e-templates/snippet-generation.yml]: right, sorry I got confused with the tests we were running yesterday on the apidoctor pipelines and we ended up back on windows. Fixed. Thanks for catching that!

**Comment by baywet** [comment]
FYI the end 2 end pipelines are now passing https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=34807&view=results

**Comment by zengin** [comment]
I didn't see a run for V1 [E2E](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build?definitionId=72), so I queued one.

**Comment by baywet** [comment]
Did you set the resources to the right branches? (code-generator and apidoctor to feature/dotnet-core)

**Comment by zengin** [comment]
I did

File: scripts/generateCSharpModels.ps1 [hunk 3]
```diff
@@ -25,17 +26,19 @@ Write-Host "Transformed $metadataFile to $cleanMetadataFile" -ForegroundColor Gr
 
 # run Typewriter
 Write-Host "Running Typewriter..."
+Push-Location -Path $typewriterDirectory
 & $typewriterPath -v Info -m $cleanMetadataFile -o $outputPath -g Files
+Pop-Location
```
Comment (zengin) [inline scripts/generateCSharpModels.ps1:31]: Q: Since the file paths are absolute, do we still need to go into typewriterDirectory?

**Review by zengin — APPROVED** [review]

File: scripts/generateCSharpModels.ps1 (hunk 3 above)
Comment (baywet) [reply scripts/generateCSharpModels.ps1:31]: typewriter itself looks relatively for the configuration files (the things in the .config folder), so effectively if the working directory is wrong, it won't find the files

Comment (zengin) [reply scripts/generateCSharpModels.ps1:31]: I see. Did we change that behavior in Typewriter code? Otherwise it would have failed earlier too, right?

**Comment by zengin** [comment]
@baywet I should have warned, but merging this will now break the daily staging runs because Typewriter changes are not yet in.

**Comment by baywet** [comment]
almost there on typewriter :)

**Pull request merged**